### ISSUE TITLE
Can not start (cleanly) shutted down node in PXC

### ISSUE THREAD

**Issue by zarere** [datacharmer/dbdeployer#86]
**Describe the bug**
After starting successfully the 3 node pxc with **start_all**  I stopped one node(node1) and after trying to start the same node while the other 2 (node2/3) are up the node1 was unable to start.

**To Reproduce**
1. Run the command` /media/zarere/57f67066-a645-4244-8359-16c191fbadc1/sandboxes/pxc_msb_5_7_25/start_all` 
2. Run the command` /media/zarere/57f67066-a645-4244-8359-16c191fbadc1/sandboxes/pxc_msb_5_7_25/node1/stop`
3. Run command `/media/zarere/57f67066-a645-4244-8359-16c191fbadc1/sandboxes/pxc_msb_5_7_25/node1/start`
4.See error in node1 err log

```
2019-06-28T08:08:52.538513Z 0 [ERROR] WSREP: It may not be safe to bootstrap the cluster from this node. It was not the last one to leave the cluster and may not contain all the updates. To force cluster bootstrap with this node, edit the grastate.dat file manually and set safe_to_bootstrap to 1 .
2019-06-28T08:08:52.538519Z 0 [ERROR] WSREP: Provider/Node (gcomm://) failed to establish connection with cluster (reason: 7)
2019-06-28T08:08:52.538523Z 0 [ERROR] Aborting
```
5.It seems it tries to bootstrap node1 instead of normal start it without bootstrap

**Expected behavior**
Node1 to start normally and join the cluster(node2/3)

**NO Screenshots**

```
$ /media/zarere/57f67066-a645-4244-8359-16c191fbadc1/sandboxes/pxc_msb_5_7_25/start_all 
# executing 'start' on /media/zarere/57f67066-a645-4244-8359-16c191fbadc1/sandboxes/pxc_msb_5_7_25
executing "start" on node 1
.. sandbox server started
executing "start" on node 2
.... sandbox server started
executing "start" on node 3
... sandbox server started
```

```
zarere@zarere-ThinkPad-W520:~$ /media/zarere/57f67066-a645-4244-8359-16c191fbadc1/sandboxes/pxc_msb_5_7_25/node1/stop 
stop /media/zarere/57f67066-a645-4244-8359-16c191fbadc1/sandboxes/pxc_msb_5_7_25/node1
```

```
zarere@zarere-ThinkPad-W520:~$ /media/zarere/57f67066-a645-4244-8359-16c191fbadc1/sandboxes/pxc_msb_5_7_25/node1/status 
node1 off
```
```
zarere@zarere-ThinkPad-W520:~$ /media/zarere/57f67066-a645-4244-8359-16c191fbadc1/sandboxes/pxc_msb_5_7_25/node1/start
................................................................................................................................................................................... sandbox server not started yet
```

**Environment:**
 - OS:  **Ubuntu 18.04.2 LTS**
 - dbdeployer version **1.33.0**
 - tarball full name **Percona-XtraDB-Cluster-5.7.25-rel28-31.35.1.Linux.x86_64.ssl100.tar.gz**
 - tarball origin **https://www.percona.com/downloads/Percona-XtraDB-Cluster-LATEST/**

**Hardware: **

 - Free storage 
```
 df -h /media/zarere/57f67066-a645-4244-8359-16c191fbadc1
Filesystem                                             Size  Used Avail Use% Mounted on
/dev/mapper/luks-aa641c06-bbf6-4f38-a781-6cd33d7b5fa9  146G  4.9G  134G   4% /media/zarere/57f67066-a645-4244-8359-16c191fbadc1
```
 - Total RAM **12GB**

**Comment by datacharmer** [comment]
I am not an expert in PXC. I was told by Percona engineers that the correct way of shutting down the cluster is in the inverse order of its start-up.
dbdeployer starts the nodes in order 1-2-3, and stops them in order 3-2-1.

When the cluster is up, if you stop node n. 3 and restart it, all is well. Stopping node 1 you get what you have reported.
The reason for this behavior is that each node has a different configuration, and the server reacts differently depending on what is in the conf file.

You should ask Percona about how to handle this situation.

The good thing is that it is reproducible!

**Comment by zarere** [comment]
I'm aware how to make it work again but just wanted to share that behavior since in that case you can stop only node 2/3 and not  node1 since somewhere in the code of dbdeployer node1 has been selected to be bootstraped after each start.The correct way in  bootstraping a PXC is - the last node that has been stopped to be started first and in case where node1 is not stopped last this error will occur.

Anyway thanks for your work on that great software and keep up with the good work.

**Comment by datacharmer** [comment]
Could someone suggest how to enhance PXC replication handling to avoid this problem?
@askdba, perhaps?

**Comment by rameshvs02** [comment]
@zarere As per the current design, PXC cluster node addresses(wsrep_cluster_address) are not added in configuration files. It is added only on the node3 configuration file. If we add the complete cluster address on all config files we can resolve this issue. 
eg:
```$ grep wsrep_cluster_address */*cnf
node1/my.sandbox.cnf:wsrep_cluster_address=gcomm://127.0.0.1:27945,127.0.0.1:27947,127.0.0.1:27949
node2/my.sandbox.cnf:wsrep_cluster_address=gcomm://127.0.0.1:27945,127.0.0.1:27947,127.0.0.1:27949
node3/my.sandbox.cnf:wsrep_cluster_address=gcomm://127.0.0.1:27945,127.0.0.1:27947,127.0.0.1:27949
$

**Comment by datacharmer** [comment]
> If we add the complete cluster address on all config files we can resolve this issue.

I disagree. If we add the complete address to all nodes, the cluster won't start.

Try this:

1. 
```
dbdeployer deploy replication --topology=pxc --concurrent pxc8.0 --skip-start
```
Now the nodes are deployed, but not started.
2. Edit the configuration files, so that all three nodes have the complete address.
3. Start the nodes (`~/sandboxes/pxc_msb_pxc8_0_19/start_all`). 
4. The server start will fail. Check the error log and you will see something like this
```
[ERROR] [MY-000000] [Galera] failed to open gcomm backend connection: 131: No address to connect (FATAL)
onnect_precheck():311
[ERROR] [MY-000000] [Galera] gcs/src/gcs_core.cpp:gcs_core_open():220: Failed to open backend connection: -131 (State not recoverable)
[ERROR] [MY-000000] [Galera] gcs/src/gcs.cpp:gcs_open():1694: Failed to open channel 'my_wsrep_cluster' at 'gcomm://127.0.0.1:27945,gcomm://127.0.0.1:27947,gcomm://127.0.0.1:27949': -131 (State not recoverable)
[ERROR] [MY-000000] [Galera] gcs connect failed: State not recoverable
[ERROR] [MY-000000] [WSREP] Provider/Node (gcomm://127.0.0.1:27945,gcomm://127.0.0.1:27947,gcomm://127.0.0.1:27949) failed to establish connection with cluster (reason: 7)
[ERROR] [MY-010119] [Server] Aborting
```

**Comment by rameshvs02** [comment]
@datacharmer To star the cluster we need to add bootstrapping variable`--wsrep-new-cluster` in `node1/start`.  Variable `--wsrep-new-cluster` is only needed for bootstrapping the cluster. We should add a condition to check the cluster state before adding the bootstrapping variable. We can restart any node without bootstrapping variable.

```
$ grep wsrep-new-cluster node1/start
    $MYSQLD_SAFE --defaults-file=$SBDIR/my.sandbox.cnf --wsrep-new-cluster $CUSTOM_MYSQLD $@ > "$SBDIR/start.log" 2>&1 &
$

$ ./start_all
# executing 'start' on /home/ramesh/sandboxes/pxc_msb_pxc8_0_18
executing "start" on node 1
... sandbox server started
executing "start" on node 2
..... sandbox server started
executing "start" on node 3
..... sandbox server started
$

Restarting node2

$ ./node2/restart
stop /home/ramesh/sandboxes/pxc_msb_pxc8_0_18/node2
...... sandbox server started
$

Checking cluster status after restart.

$ ./check_nodes
# Node 1
+---------------------------+-------------------------------------------------+
| Variable_name             | Value                                           |
+---------------------------+-------------------------------------------------+
| wsrep_local_state_uuid    | c51c0936-7807-11ea-af6b-c2e8fb959a17            |
| wsrep_received            | 14                                              |
| wsrep_local_commits       | 0                                               |
| wsrep_local_state         | 4                                               |
| wsrep_local_state_comment | Synced                                          |
| wsrep_incoming_addresses  | 127.0.0.1:27819,127.0.0.1:27821,127.0.0.1:27820 |
| wsrep_gcomm_uuid          | 530caf03-7808-11ea-bc31-2e42ddd945a2            |
| wsrep_cluster_size        | 3                                               |
| wsrep_cluster_status      | Primary                                         |
| wsrep_connected           | ON                                              |
| wsrep_ready               | ON                                              |
+---------------------------+-------------------------------------------------+
# Node 2
+---------------------------+-------------------------------------------------+
| Variable_name             | Value                                           |
+---------------------------+-------------------------------------------------+
| wsrep_local_state_uuid    | c51c0936-7807-11ea-af6b-c2e8fb959a17            |
| wsrep_received            | 3                                               |
| wsrep_local_commits       | 0                                               |
| wsrep_local_state         | 4                                               |
| wsrep_local_state_comment | Synced                                          |
| wsrep_incoming_addresses  | 127.0.0.1:27819,127.0.0.1:27821,127.0.0.1:27820 |
| wsrep_gcomm_uuid          | 7a4b7d7e-7809-11ea-8650-4bbeed50e4e1            |
| wsrep_cluster_size        | 3                                               |
| wsrep_cluster_status      | Primary                                         |
| wsrep_connected           | ON                                              |
| wsrep_ready               | ON                                              |
+---------------------------+-------------------------------------------------+
# Node 3
+---------------------------+-------------------------------------------------+
| Variable_name             | Value                                           |
+---------------------------+-------------------------------------------------+
| wsrep_local_state_uuid    | c51c0936-7807-11ea-af6b-c2e8fb959a17            |
| wsrep_received            | 13                                              |
| wsrep_local_commits       | 0                                               |
| wsrep_local_state         | 4                                               |
| wsrep_local_state_comment | Synced                                          |
| wsrep_incoming_addresses  | 127.0.0.1:27819,127.0.0.1:27821,127.0.0.1:27820 |
| wsrep_gcomm_uuid          | 57d44837-7808-11ea-8ab0-6a8412893eea            |
| wsrep_cluster_size        | 3                                               |
| wsrep_cluster_status      | Primary                                         |
| wsrep_connected           | ON                                              |
| wsrep_ready               | ON                                              |
+---------------------------+-------------------------------------------------+
$

$ grep wsrep_cluster_address */*cnf
node1/my.sandbox.cnf:wsrep_cluster_address=gcomm://127.0.0.1:27945,127.0.0.1:27947,127.0.0.1:27949
node2/my.sandbox.cnf:wsrep_cluster_address=gcomm://127.0.0.1:27945,127.0.0.1:27947,127.0.0.1:27949
node3/my.sandbox.cnf:wsrep_cluster_address=gcomm://127.0.0.1:27945,127.0.0.1:27947,127.0.0.1:27949

$

**Comment by datacharmer** [comment]
@rameshvs02 do we need the option `--wsrep-new-cluster` also when we restart node 1 alone?

**Comment by rameshvs02** [comment]
@datacharmer `--wsrep-new-cluster` is not required to restart node1. `--wsrep-new-cluster` is only required for bootstrapping the first node. We can add `--wsrep-new-cluster` checking condition to all nodes. This will help to bootstrap the cluster from any node.

**Comment by datacharmer** [comment]
@rameshvs02 The difficult thing here is that the `start` script is created at deployment time for all nodes, but we want to use it in different ways. If we run things manually, it would be fine, but `dbdeployer` goal is to avoid doing things manually as much as possible. Thus, I need to figure out how to design this script to do these two different things, or define a separate script.

**Comment by rameshvs02** [comment]
@datacharmer it would be great if we can automate this!

**Comment by datacharmer** [comment]
Fixed in [v1.49.0](https://github.com/datacharmer/dbdeployer/releases/tag/v1.49.0)
